src/core/core_utils.js

```diff
@@ -287,19 +287,26 @@ function _collectJS(entry, xref, list, parents) {
 
 function collectActions(xref, dict, eventType) {
   const actions = Object.create(null);
-  if (dict.has("AA")) {
-    const additionalActions = dict.get("AA");
-    for (const key of additionalActions.getKeys()) {
-      const action = eventType[key];
-      if (!action) {
-        continue;
-      }
-      const actionDict = additionalActions.getRaw(key);
-      const parents = new RefSet();
-      const list = [];
-      _collectJS(actionDict, xref, list, parents);
-      if (list.length > 0) {
-        actions[action] = list;
+  const additionalActionsDicts = getInheritableProperty({
+    dict,
+    key: "AA",
+    stopWhenFound: false,
+  });
+  if (additionalActionsDicts) {
+    for (let i = additionalActionsDicts.length - 1; i >= 0; i--) {
+      const additionalActions = additionalActionsDicts[i];
```

Comment:
*Most a pre-existing issue, but:* You probably want to also wrap the loop below in a `if (additionalActions instanceof Dict) { ... }`-check to avoid throwing if the data isn't a Dictionary (e.g. if the PDF is somehow corrupt).